### DISCUSSION TITLE
Beta-barrel ab initio

### DISCUSSION THREAD

**Discussion by gserci** [RosettaCommons/rosetta#78] [General]
Hello everyone,

I have a protein consisting of 332 aminoacids. This protein is a mutant form of another protein with 349 aminoacids (there is a deletion of 17 aminoacids), which is a beta-barrel with a known structure. I want to predict the structure of the mutant protein. I tried using Rosetta CM with the original protein as a template, but I'm not satisfied with the results, so I want to try predicting the structure using ab initio methods. I tried using Rosetta ab initio with the default options, but the protein may be too large, and the results were very poor. Knowing that the protein is a 16-strand beta-barrel, are there any parameters you suggest using to achieve better predictions with the ab initio method?

Thank you.

**Comment by roccomoretti** [comment]
You're correct in surmising that 349 aa is too big for Rosetta ab initio. (Additionally, Rosetta ab initio tends to have more difficulty with beta strands than alpha helicies, given the need for long-range contacts.)

That said, not many people use Rosetta ab initio these days -- for most of what it does, AlphaFold or RoseTTAFold does things better, and often with less compute. Hopefully running them in "vanilla" mode works for your system, but if it doesn't, I believe there's ways to do fold-conditioning, explicitly telling them about structural details. . For example, that it's a 16-strand beta-barrel -- though I would imagine that simply using the homologous protein as a template would cause either to pick that up.

Regarding poor results with RosettaCM, note that it's rather sensitive to alignments. Depending on what issues you're having with the results, you might get better results by manually tweaking the alignments. You should also be able to manually set up constraints/restraints to enforce particular geometric features in the results. Though depending on what you're doing, trying AlphaFold/RoseTTAFold first might be recommended.

**Reply by gserci** [reply]
Thank you very much for your reply. AlphaFold 3 and RosettaFold both give me the same structure. Using the template with ColabFold (AlphaFold 2) and RosettaCM, the structure is again the same as AlphaFold 3 and RosettaFold. However, I think this doesn't make much sense because it's basically a 16-strand beta-barrel with a hole, and the scoring for the loop in the deletion zone is less than 50%. I can construct a structure on my own with Assembler, but this method is totally arbitrary, and I don't know how to calculate the stability of the structure I've built. 🤷🏻‍♂️

**Reply by roccomoretti** [reply]
If you're getting consistent results with all the different structure prediction programs, it might be worth taking a step back and figuring out what about the model you're dissatisfied with. Depending on what you're after, you may be able to do some remodeling/resampling (post-processing) of the predicted results to rebuild those regions which aren't modeled so well. (For example, [IterativeHybridize](https://docs.rosettacommons.org/docs/latest/IterativeHybridize) has been found to improve RosettaCM modeling. You can certainly attempt that with judiciously filtered/trimmed template models to rebuild those sections you're dissatisfied.) You might alternatively be able to add restraints or do fold conditioning on the original prediction model to promote the specific features which you want.

I'll additionally note that there are several procedures out there to get conformational diversity out of machine learning models (typically AlphaFold2). [Sala et al.](https://pubmed.ncbi.nlm.nih.gov/37392556/) is one recent review. You might try some of those approaches to see if the alternate states they pick up match your conceptions better.

**Reply by gserci** [reply]
Thank you so much, I'll try applying your suggestions!

**Comment by NooraAz** [comment]
using `-abinitio:steal_Xmer` flag might help a bit, but as Rocco mentioned, using ab initio is not recommended these days and it probably won't give a satisfactory result. 

About Alphafold, I suggest getting the MSA separately, removing high-score MSA data, and running it only using shallow MSAs, as suggested in the mentioned paper. But generally speaking, Alphafold is not great at predicting such small changes. So it is probably a good idea to consider a coarse-grained MD as well.

**Comment by roccomoretti** [comment]
Marking this as "closed", as I hope your question has been answered. If that assessment is incorrect, feel free to re-open or open up a new discussion with more details.
